src/compile/data/impute.ts

```diff
@@ -1,14 +1,92 @@
-import {normalizeimputeProperties} from '../../impute';
+import {isArray} from 'vega-util';
+import {Encoding} from '../../encoding';
+import {getFieldDef, isFieldDef, vgField} from '../../fielddef';
 import {ImputeSequence, ImputeTransform, isImputeSequence} from '../../transform';
-import {duplicate} from '../../util';
+import {duplicate, keys} from '../../util';
 import {VgFormulaTransform, VgImputeTransform, VgSignalRef, VgWindowTransform} from '../../vega.schema';
 import {UnitModel} from '../unit';
 import {DataFlowNode} from './dataflow';
 
-// TODO Rename to ImputeNode
-export class ImputeTransformNode extends DataFlowNode {
+export function normalizeimputeProperties(encoding: Encoding<string>): ImputeTransform | undefined {
```

Comment:
normalizeImputeProperties

Reply:
Also, https://github.com/vega/vega-lite/pull/3881/files/08844a9f23a0eaa158969ebe538e7a841367de7c..ddd334dd544ca99792df7648c2537e08783604ba#r195305085

Reply:
Ah I get what do you now.

Reply:
Instead of having a separate function here why don't you copy all the code in this method to be in `makeFromEncoding`?  I don't see why we need an additional code jump here.